libcxx/test/support/test_comparisons.h

```diff
@@ -268,6 +268,70 @@ struct PartialOrder {
   }
 };
 
-#endif
+template <typename T1, typename T2 = T1>
```

Comment:
I guess the dependency on the previous PR was due to conflicts in test files. I wonder whether we can make the same additions to the common test files in both PRs, and drop the changes in the the PR merged later.

Reply:
Basically, yes. This PR has everything in it. It's just that some configurations in the CI will fail because of the missing refactoring from https://github.com/llvm/llvm-project/pull/139368 rebasing the current PR on the latter will fix those.